src/components/ui/separator.tsx

```diff
@@ -0,0 +1,28 @@
+"use client"
+
+import * as React from "react"
+import * as SeparatorPrimitive from "@radix-ui/react-separator"
+
+import { cn } from "@/lib/utils"
+
+function Separator({
+  className,
+  orientation = "horizontal",
+  decorative = true,
+  ...props
+}: React.ComponentProps<typeof SeparatorPrimitive.Root>) {
+  return (
+    <SeparatorPrimitive.Root
+      data-slot="separator"
+      decorative={decorative}
+      orientation={orientation}
+      className={cn(
+        "bg-border shrink-0 data-[orientation=horizontal]:h-px data-[orientation=horizontal]:w-full data-[orientation=vertical]:h-full data-[orientation=vertical]:w-px",
+        className
+      )}
+      {...props}
+    />
+  )
+}
+
+export { Separator }
```

Comment:
Lot of code for a separator lol. But that's okay if you're trying to achieve a specific look.

That does remind me of something I was going to say when I finished submitting the review though... For PRs like this with highly visual elements, you want to include screenshots of all the new/redesigned pages so I don't need to open the app to see what this all looks like while I'm deep in reviewing the code. It helps to understand how all these components come together.